### PR TITLE
Feature: support calculated field

### PR DESCRIPTION
**Pull request by onlyjackfrost**
Support calculated field in sementic layer

### CODE REVIEW

File: wren-ui/migrations/20240430033014_update_model_column_table.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
```
Comment (wwwy3y3) [inline wren-ui/migrations/20240430033014_update_model_column_table.js:1]: @onlyjackfrost could you add some comments here on why we're taking out this column? thx

File: wren-ui/src/apollo/server/adaptors/wrenEngineAdaptor.ts [hunk 2]
```diff
@@ -47,6 +47,25 @@ export interface DescribeStatementResponse {
   columns: ColumnMetadata[];
 }
 
+export enum WrenEngineValidateStatusEnum {
```
Comment (wwwy3y3) [inline wren-ui/src/apollo/server/adaptors/wrenEngineAdaptor.ts:50]: nit: feel like `Enum` postfix is unnecessary. Naming could be `WrenEngineValidationStatus`.

File: wren-ui/src/apollo/server/mdl/mdlBuilder.ts [hunk 3]
```diff
@@ -118,46 +119,114 @@ export class MDLBuilder implements IMDLBuilder {
     });
   }
 
-  public addColumn(): void {
+  public addNormalField(): void {
```
Comment (wwwy3y3) [inline wren-ui/src/apollo/server/mdl/mdlBuilder.ts:122]: Good naming change to reflect the difference between columns and calculated fields.

**Review by wwwy3y3 — APPROVED** [review]
Others LGTM

File: wren-ui/migrations/20240430033014_update_model_column_table.js (hunk 1 above)
Comment (onlyjackfrost) [reply wren-ui/migrations/20240430033014_update_model_column_table.js:1]: delivered in 9ec6f9620239b869a9716daf8c6827d85c23995a

File: wren-ui/src/apollo/server/adaptors/wrenEngineAdaptor.ts (hunk 2 above)
Comment (onlyjackfrost) [reply wren-ui/src/apollo/server/adaptors/wrenEngineAdaptor.ts]: delivered in 9ec6f9620239b869a9716daf8c6827d85c23995a

**Pull request merged**